aws/resource_aws_macie2_custom_data_identifier.go

```diff
@@ -0,0 +1,232 @@
+			},
+			"maximum_match_distance": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ForceNew:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(1, 300),
+			},
+			"tags":     tagsSchemaForceNew(),
+			"tags_all": tagsSchemaComputed(),
+			"deleted": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceMacie2CustomDataIdentifierCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).macie2conn
+
+	defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(keyvaluetags.New(d.Get("tags").(map[string]interface{})))
+
+	input := &macie2.CreateCustomDataIdentifierInput{
+		ClientToken: aws.String(resource.UniqueId()),
+	}
+
+	if v, ok := d.GetOk("regex"); ok {
+		input.Regex = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("keywords"); ok {
+		input.Keywords = expandStringList(v.([]interface{}))
+	}
+	if v, ok := d.GetOk("ignore_words"); ok {
+		input.IgnoreWords = expandStringList(v.([]interface{}))
+	}
+	input.Name = aws.String(naming.Generate(d.Get("name").(string), d.Get("name_prefix").(string)))
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("maximum_match_distance"); ok {
+		input.MaximumMatchDistance = aws.Int64(int64(v.(int)))
+	}
+	if len(tags) > 0 {
+		input.Tags = tags.IgnoreAws().Macie2Tags()
+	}
+
+	var err error
+	var output *macie2.CreateCustomDataIdentifierOutput
+	err = resource.RetryContext(ctx, 4*time.Minute, func() *resource.RetryError {
+		output, err = conn.CreateCustomDataIdentifierWithContext(ctx, input)
+		if err != nil {
+			if tfawserr.ErrCodeEquals(err, macie2.ErrorCodeClientError) {
+				return resource.RetryableError(err)
+			}
+
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if isResourceTimeoutError(err) {
+		output, err = conn.CreateCustomDataIdentifierWithContext(ctx, input)
+	}
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating Macie CustomDataIdentifier: %w", err))
+	}
+
+	d.SetId(aws.StringValue(output.CustomDataIdentifierId))
+
+	return resourceMacie2CustomDataIdentifierRead(ctx, d, meta)
+}
+
+func resourceMacie2CustomDataIdentifierRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).macie2conn
+
+	defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+	input := &macie2.GetCustomDataIdentifierInput{
+		Id: aws.String(d.Id()),
+	}
+
+	resp, err := conn.GetCustomDataIdentifierWithContext(ctx, input)
+
+	if tfawserr.ErrCodeEquals(err, macie2.ErrCodeResourceNotFoundException) ||
+		tfawserr.ErrCodeEquals(err, macie2.ErrCodeAccessDeniedException) {
```

Comment:
We'll want to differentiate between when we get `macie2.ErrCodeAccessDeniedException` because the user doesn't have access vs Macie is not enabled. In the other resources, we've checked

```suggestion
		tfawserr.ErrMessageContains(err, macie2.ErrCodeAccessDeniedException, "Macie is not enabled") {
```